internal/slicer/slicer_test.go

```diff
@@ -53,294 +53,292 @@ var packageEntries = map[string][]testutil.TarEntry{
 	summary: "Basic slicing",
-	slices:  []setup.SliceKey{{"base-files", "myslice"}},
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
 	release: map[string]string{
-		"slices/mydir/base-files.yaml": `
-			package: base-files
+		"slices/mydir/test-package.yaml": `
+			package: test-package
 			slices:
 				myslice:
 					contents:
-						/usr/bin/hello:
-						/usr/bin/hallo: {copy: /usr/bin/hello}
-						/bin/hallo:     {symlink: ../usr/bin/hallo}
-						/etc/passwd:    {text: data1}
-						/etc/dir/sub/:  {make: true, mode: 01777}
+						/dir/file:
+						/dir/file_copy:  {copy: /dir/file}
+						/other_dir/file: {symlink: ../dir/file}
+						/dir/text_file:  {text: data1}
+						/dir/foo/bar/:   {make: true, mode: 01777}
 		`,
 	},
 	result: map[string]string{
-		"/usr/":          "dir 0755",
-		"/usr/bin/":      "dir 0755",
-		"/usr/bin/hello": "file 0775 eaf29575",
-		"/usr/bin/hallo": "file 0775 eaf29575",
-		"/bin/":          "dir 0755",
-		"/bin/hallo":     "symlink ../usr/bin/hallo",
-		"/etc/":          "dir 0755",
-		"/etc/dir/":      "dir 0755",
-		"/etc/dir/sub/":  "dir 01777",
-		"/etc/passwd":    "file 0644 5b41362b",
+		"/dir/":           "dir 0755",
+		"/dir/file":       "file 0644 cc55e2ec",
+		"/dir/file_copy":  "file 0644 cc55e2ec",
+		"/dir/foo/":       "dir 0755",
+		"/dir/foo/bar/":   "dir 01777",
+		"/dir/text_file":  "file 0644 5b41362b",
+		"/other_dir/":     "dir 0755",
+		"/other_dir/file": "symlink ../dir/file",
 	},
 }, {
 	summary: "Glob extraction",
-	slices:  []setup.SliceKey{{"base-files", "myslice"}},
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
 	release: map[string]string{
-		"slices/mydir/base-files.yaml": `
-			package: base-files
+		"slices/mydir/test-package.yaml": `
+			package: test-package
 			slices:
 				myslice:
 					contents:
-						/**/he*o:
+						/**/other_f*e:
 		`,
 	},
 	result: map[string]string{
-		"/usr/":          "dir 0755",
-		"/usr/bin/":      "dir 0755",
-		"/usr/bin/hello": "file 0775 eaf29575",
+		"/dir/":                  "dir 0755",
+		"/dir/nested/":           "dir 0755",
+		"/dir/nested/other_file": "file 0644 6b86b273",
+		"/dir/other_file":        "file 0644 63d5dd49",
 	},
 }, {
-	summary: "Create new file under extracted directory",
-	slices:  []setup.SliceKey{{"base-files", "myslice"}},
+	summary: "Create new file under extracted directory and preserve parent directory permissions",
+	slices:  []setup.SliceKey{{"test-package", "myslice"}},
 	release: map[string]string{
-		"slices/mydir/base-files.yaml": `
-			package: base-files
+		"slices/mydir/test-package.yaml": `
+			package: test-package
 			slices:
 				myslice:
 					contents:
-						# Note the missing /tmp/ here.
-						/tmp/new: {text: data1}
+						# Note the missing parent directories here.
+						/parent/new: {text: data1}
 		`,
 	},
 	result: map[string]string{
-		"/tmp/":    "dir 01777", // This is the magic.
-		"/tmp/new": "file 0644 5b41362b",
+		"/parent/":    "dir 01777",
```

Comment:
Feel free to reword the comment to be more obvious, but the reason it was here was to highlight that this is the main point of the test, so it why would we drop it?

Reply:
I felt the new summary was making it clear. There is no downside to having it though, so I will re-add it.